module/VuFind/src/VuFind/Controller/InstallController.php

```diff
@@ -922,4 +923,57 @@ public function homeAction()
         }
         return $this->createViewModel(['checks' => $checks]);
     }
+
+    /**
+     * Get minimal PHP version required for VuFind to run.
+     *
+     * @return string
+     */
+    protected function getMinimalPhpVersion(): string
+    {
+        $composer = $this->getComposerJson();
+        if (empty($composer)) {
+            throw new \Exception('Cannot find composer.json');
+        }
+        $rawVersion = $composer['require']['php'] ?? '';
+        $version = preg_replace('/[^0-9. ]/', '', $rawVersion);
+        $version = preg_split('/[. ]/', $version);
+        if (count($version) >= 3) {
+            return sprintf('%d.%d.%d', $version[0], $version[1], $version[2]);
+        }
+        $version = $composer['config']['platform']['php'] ?? '';
```

Comment:
Why do you do validation/normalization on `$composer['require']['php']` but not on `$composer['config']['platform']['php']`? Would it be better to just check both locations in the assignment to `$rawVersion` prior to normalization? I realize that one field may have special characters that the other does not, but I don't think it hurts to normalize both just to be safe.

Reply:
Yes, that was the reason. But as you say, it does not hurt and simplifies the code a bit so I've changed it as you suggested.